src/ds1307/ds1307.cxx

```diff
@@ -0,0 +1,207 @@
+/*
+ * Author: Jon Trulson <jtrulson@ics.com>
+ * Copyright (c) 2014 Intel Corporation.
+ *
+ * Adapted from Seeed Studio library:
+ * https://github.com/Seeed-Studio/RTC_DS1307
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining
+ * a copy of this software and associated documentation files (the
+ * "Software"), to deal in the Software without restriction, including
+ * without limitation the rights to use, copy, modify, merge, publish,
+ * distribute, sublicense, and/or sell copies of the Software, and to
+ * permit persons to whom the Software is furnished to do so, subject to
+ * the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be
+ * included in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+ * LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+ * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+ * WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+#include <iostream>
+#include <string>
+
+#include "ds1307.h"
+
+using namespace upm;
+using namespace std;
+
+
+DS1307::DS1307(int bus)
+{
+  // setup our i2c link
+  m_i2c = mraa_i2c_init(bus);
+
+  mraa_result_t ret = mraa_i2c_address(m_i2c, DS1307_I2C_ADDR);
+
+  if (ret != MRAA_SUCCESS) 
+    cerr << "DS1307: Could not initialize i2c bus. " << endl;
+}
+
+DS1307::~DS1307()
+{
+  mraa_i2c_stop(m_i2c);
+}
+
+mraa_result_t DS1307::writeBytes(uint8_t reg, uint8_t *buffer, unsigned int len)
+{
+  if (!len || !buffer)
+    return MRAA_SUCCESS;
```

Comment:
For bad input (0-length or null buffer), would it be better to return MRAA_INVALID_PARAMETER to alert the user?  If you agree, I can make the change when merging.  Thanks!